walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,194 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long meetingDurationMinutes = request.getDuration();
+
+    // If the meeting duration is over 24 hours, then return no available timeRange
+    if (meetingDurationMinutes > TimeRange.WHOLE_DAY.duration()) {
+      return Collections.emptyList();
+    }
+
+    // If no Attendees are passed in, return a timeRange containing the whole day
+    if (request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+   
+   // Store unavailable TimeRanges for mandatory and optional attendees separately
+    List<TimeRange> unavailableMandatoryTimeRanges = getUnavailableTimeRanges(
+        events, mandatoryAttendees);
+    List<TimeRange> unavailableOptionalTimeRanges = getUnavailableTimeRanges(events, optionalAttendees);
+
+    // Get available TimeRanges from unavailable TimeRanges
+    List<TimeRange> availableMandatoryTimeRanges = getAvailableTimeRanges(
+        unavailableMandatoryTimeRanges, meetingDurationMinutes);
+    List<TimeRange> availableOptionalTimeRanges = getAvailableTimeRanges(
+        unavailableOptionalTimeRanges, meetingDurationMinutes);
+
+    // If no mandatory attendees are passed, return the available TimeRanges for optional attendees
+    // else if no optional attendees are passed, or no TimeRanges are available for optiona attendees
+    // then return available TimeRanges for mandatory attendees, else merge the available TimeRanges
+    // for mandatory attendees with available TimeRanges for optional employees
+    if (mandatoryAttendees.isEmpty()) {
+      return availableOptionalTimeRanges;
+    } else if (optionalAttendees.isEmpty() || availableOptionalTimeRanges.isEmpty()) {
+      return availableMandatoryTimeRanges;
+    } else {
+      return mergeAvailableTimeRanges(
+          availableMandatoryTimeRanges, availableOptionalTimeRanges, meetingDurationMinutes);
+    }
+}
```

Comment:
Indentation seems off. I think we need two spaces. And we need a line break below.

Reply:
I thought line breaking was 4 spaces.
https://google.github.io/styleguide/javaguide.html#s4.5-line-wrapping